src/devices/Pigpio/Driver.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Pigpio
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class Driver : GpioDriver
+    {
+        private readonly IPEndPoint _endpoint;
+        private readonly PigpiodIf _proxy;
+        private readonly List<int> _openPins;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="endpoint"></param>
+        public Driver(IPEndPoint endpoint)
+        {
+            _endpoint = endpoint;
+            _proxy = new PigpiodIf();
+
+            _openPins = new List<int>();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        public Task ConnectAsync()
```

Comment:
would it make sense to require being connected to have a valid instance of the class? I.e.:
`GpioDriver driver = await PigPiGpioDriver.ConnectAsync(...);`

Can't see how this class is useful otherwise and API would feel cleaner (constructor could be private/protected)